lib/event_bus/services/observation.ex

```diff
@@ -44,19 +44,25 @@ defmodule EventBus.Service.Observation do
   @doc false
   @spec mark_as_completed(subscriber_with_event_ref()) :: :ok
   def mark_as_completed({subscriber, event_shadow}) do
-    {subscribers, completers, skippers} = fetch(event_shadow)
-    save_or_delete(event_shadow, {subscribers, [subscriber | completers], skippers})
+    case fetch(event_shadow) do
+      {subscribers, completers, skippers} ->
+        save_or_delete(event_shadow, {subscribers, [subscriber | completers], skippers})
+        nil -> :ok
```

Comment:
In here, we are loosing the context of why we are getting `nil` from the `ETS` storage.

Reply:
Maybe, handling on the lower level could provide a better idea?

Reply:
In `fetch`, if the data is missing, `ETS.lookup` returns an `[]` which doesn't have any contextual information. I could add the logging back again.

```
 case Ets.lookup(table_name(topic), id) do
      [{_, data}] -> data
      _ -> nil     # <-- Can add the logging here again?
    end
```